src/services/User.service.js

```diff
@@ -0,0 +1,35 @@
+const { generateId } = require('../utils.js');
+let users = [];
+
+const findUserById = (id) => {
+  return users.find((usr) => usr.id === id);
+};
+
+const updateUserRoom = (id, room) => {
+  const user = findUserById(id);
+
+  user.currentRoom = room;
+
+  return user;
+};
+
+const removeOne = (id) => {
+  users = users.filter((usr) => usr.id !== id);
+};
+
+const addOne = (user) => {
+  users.push(user);
+};
+
+const buildUser = (username) => {
+  return { id: generateId(users), username };
+};
+
+module.exports = {
+  updateUserRoom,
+  findUserById,
+  removeOne,
+  addOne,
+  buildUser,
+  users,
```

Comment:
Instead of exporting a variable, it would be recommended to use a getter method, like `getUsers()` that return the users variable. It's not good practice to export directly the variable between components.